baybe/recommenders/pure/base.py

```diff
@@ -212,10 +245,7 @@ def _recommend_with_discrete_parts(
         if (not is_hybrid_space) and (len(candidates_exp) < batch_size):
             raise NotEnoughPointsLeftError(
                 f"Using the current settings, there are fewer than {batch_size} "
-                "possible data points left to recommend. This can happen "
-                "when all candidates have already been measured/recommended while "
-                "`allow_repeated_recommendations'/'allow_recommending_already_measured' "  # noqa: E501
-                "are set to `False`."
```

Comment:
True. But having it here in the recommender could potentially cause more harm than keeping it. Because when someone calls the recommender directly (i.e. where the flags don't even exist) then the hint would point the user to something they wouldn't be able to change, probably causing confusion.

Have a look at the new version and let me know what you think. It required a few lines of code, but now the user gets a tailored recommendation for their situation, depending on the actual flag values and whether or not stuff is called from campaign/recommender.

Reply:
ty seems reasonable